### PR TITLE
[feat] embed openapi spec

### PR DESCRIPTION
**Pull request by SteveHNH**
Rather than reading from the filesystem every time we want to present
the api spec. We can embed the spec in the executable.

This required moving api into a path where main.go is. The 'embed' lib
cannnot read relative paths so it makes it necessary to put it where the
path will be consistent

RHCLOUD-16031

Signed-off-by: Stephen Adams <tsadams@gmail.com>

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation

- [ ] Output Encoding

- [ ] Authentication and Password Management

- [ ] Session Management

- [ ] Access Control

- [ ] Cryptographic Practices

- [ ] Error Handling and Logging

- [ ] Data Protection

- [ ] Communication Security

- [ ] System Configuration

- [ ] Database Security

- [ ] File Management

- [ ] Memory Management

- [ ] General Coding Practices

### CODE REVIEW

**Comment by SteveHNH** [comment]
We're gonna need a different builder image for this one. The `embed` lib requires golang 1.16 or greater

File: internal/api/api.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package api
+
+import (
+	_ "embed"
+)
+
+//go:embed openapi.json
+var Api []byte
```
Comment (dehort) [inline internal/api/api.go:8]: This is nitpicky but I might change the name of this variable to ```ApiSpec``` or something a bit more descriptive.  I'm not sure Api communicates that its really the contents of the openapi spec file.  Just my $0.02...

**Review by dehort — APPROVED** [review]

File: internal/api/api.go (hunk 1 above)
Comment (dehort) [reply internal/api/api.go:8]: BTW, the ```embed``` approach here is pretty slick!  :+1:   :)

**Review by aleccohan — APPROVED** [review]
LGTM!

File: internal/api/api.go (hunk 1 above)
Comment (SteveHNH) [reply internal/api/api.go:8]: sweet. I'll do that. That's smart

**Pull request merged**